src/databroker_elasticsearch/elasticindex.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python3
+
+"""\
+Class for convenient access to Elasticsearch index.
+"""
+
+from typing import Callable
+from elasticsearch import Elasticsearch
+from elasticsearch import helpers as eshelpers
+
+
+class ElasticIndex:
+    """Convenience functions for adding documents to Elasticsearch index.
+
+    Parameters
+    ----------
+    es : Elasticsearch, str, or dict
+        The Elasticsearch client to push entries to.
+        When `str` or `dict` type, instantiate a new Elasticsearch object
+        using this argument.
+    index : str
+        The index to use in Elasticsearch
+    mapper : callable, optional
+        A function to transform input document to Elasticsearch entry.
+    criteria : callable, optional
+        Callable which is run on all the start document, if the return is
+        truthy the document is sent to ES else, it is not added. Defaults
+        to True for all documents
+
+    Attributes
+    ----------
+    es : Elasticsearch
```

Comment:
can this also be a string or a dict?

Reply:
Nope, as an attribute it is already converted to the ES object.
The `__init__` however takes string or dict for convenience.